### PR TITLE
Move HTML parsing to own file

### PR DESCRIPTION
**Pull request by domchristie**
This PR moves the HTML parser code, including your fixes for malformed docs and IE9, into its own file to tidy it up a bit.

Hopefully this should come through to https://github.com/domchristie/to-markdown/pull/129 once merged.

### CODE REVIEW

File: lib/html-parser.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          FetchExternalResources: [],
+          ProcessExternalResources: false
+        }
+      });
+    };
+  } else {
+    Parser.prototype.parseFromString = function (string) {
+      var doc;
+
+      if (!shouldUseActiveX()) {
```
Comment (Sebmaster) [inline lib/html-parser.js]: This is a bit less efficient since it impacts runtime performance.

Comment (domchristie) [inline lib/html-parser.js]: Improved in https://github.com/domchristie/to-markdown/commit/847a644310a6d2309b71fd3c6920b5d590dc5298

File: lib/html-parser.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        doc.write(string);
+        doc.close();
+        return doc;
+      };
+    }
+  }
+  return Parser;
+}
+
+function shouldUseActiveX () {
```
Comment (Sebmaster) [inline lib/html-parser.js]: nit: space?

Comment (domchristie) [inline lib/html-parser.js]: Sorry, what space?

Comment (Sebmaster) [inline lib/html-parser.js]: The one after the function name. It's inconsistent in this file.

Comment (domchristie) [inline lib/html-parser.js]: OK, cool. I quite like the standard style, and I’m planning on converting the library to that soon. For now I’ll add spaces to the other functions in this file.

Comment (domchristie) [inline lib/html-parser.js]: Should be fixed now

**Comment by Sebmaster** [comment]
Sweet, will merge in like 3 hours or so.

**Comment by Sebmaster** [comment]
Merged as 9f1383b6a34f1d691da574cf3ee11dab2cf17d78.

**Pull request closed without merging**